test/api/v3/integration/user/POST-user_purchase.test.js

```diff
@@ -0,0 +1,35 @@
+import {
+  generateUser,
+  translate as t,
+} from '../../../../helpers/api-integration/v3';
+
+describe('POST /user/purchase/:type/:key', () => {
+  let user;
+  let type = 'hatchingPotions';
+  let key = 'Base';
+
+  beforeEach(async () => {
+    user = await generateUser({
+      balance: 40,
+    });
+  });
+
+  // More tests in common code unit tests
+
+  it('returns an error when key is not provided', async () => {
+    await expect(user.post(`/user/purchase/${type}`))
+      .to.eventually.be.rejected.and.eql({
+        code: 404,
```

Comment:
this is returning a 404 because the user/purchase/${type} route without the last key parameter doesn't exist